staker/block_validator.go

```diff
@@ -807,28 +822,31 @@ validationsLoop:
 			return nil, nil
 		}
 		if currentStatus == Prepared {
-			input, err := validationStatus.Entry.ToInput()
-			if err != nil && ctx.Err() == nil {
-				v.possiblyFatal(fmt.Errorf("%w: error preparing validation", err))
-				continue
-			}
 			replaced := validationStatus.replaceStatus(Prepared, SendingValidation)
 			if !replaced {
 				v.possiblyFatal(errors.New("failed to set SendingValidation status"))
 			}
+			validatorValidationWaitToLaunchHist.Update(validationStatus.timeStampInterval())
 			validatorPendingValidationsGauge.Inc(1)
 			var runs []validator.ValidationRun
 			for _, moduleRoot := range wasmRoots {
+				input, err := validationStatus.Entry.ToInput(v.chosenValidator[moduleRoot].StylusArchs())
```

Comment:
nit: might be good to pull the chosenValidator[moduleRoot] out to a separate variable for use below too